src/domain/sales-channels/tables/product.tsx

```diff
@@ -0,0 +1,462 @@
+        prevPage={handlePrev}
+        hasNext={canNextPage}
+        hasPrev={canPreviousPage}
+      />
+    </div>
+  )
+}
+
+/**
+ * Renders product table row.
+ */
+const ProductRow = ({ row, actions }) => {
+  const product = row.original
+
+  return (
+    <Table.Row
+      color={"inherit"}
+      className={row.isSelected ? "bg-grey-5" : ""}
+      linkTo={`/a/products/${product.id}`}
+      actions={actions}
+      {...row.getRowProps()}
+    >
+      {" "}
+      {row.cells.map((cell, index) => {
+        return (
+          <Table.Cell {...cell.getCellProps()}>
+            {" "}
+            {cell.render("Cell", { index })}{" "}
+          </Table.Cell>
+        )
+      })}{" "}
+    </Table.Row>
+  )
+}
+
+type RemoveProductsPopupProps = {
+  close: () => void
+  onRemove: () => void
+  total: number
+}
+
+/**
+ * Popup for removing selected products from a sales channel.
+ */
+function RemoveProductsPopup({
+  close,
+  onRemove,
+  total,
+}: RemoveProductsPopupProps) {
+  const classes = {
+    "translate-y-1 opacity-0": !total,
+    "translate-y-0 opacity-100": total,
+  }
+
+  return (
+    <div
+      className={clsx(
+        "absolute w-full bottom-1 flex justify-center transition-all duration-200 pointer-events-none",
+        classes
+      )}
+    >
+      <div className="h-[48px] min-w-[224px] rounded-lg border shadow-toaster flex items-center justify-around gap-3 px-4 py-3 pointer-events-auto">
+        <span className="text-small text-grey-50">{total} selected</span>
+        <div className="w-[1px] h-[20px] bg-grey-20" />
+        <Button variant="danger" size="small" onClick={onRemove}>
+          Remove
+        </Button>
+        <button onClick={close} className="text-grey-50 cursor-pointer">
+          <CrossIcon size={20} />
+        </button>
+      </div>
+    </div>
+  )
+}
+
+/* **************************************** */
+/* ************** CONTAINERS ************** */
+/* **************************************** */
+
+type SalesChannelProductsTableProps = {
+  salesChannelId: string
+  showAddModal: () => void
+}
+
+/**
+ * Sales channel products table container.
+ */
+function SalesChannelProductsTable(props: SalesChannelProductsTableProps) {
+  const { salesChannelId, showAddModal } = props
+  const [selectedRowIds, setSelectedRowIds] = useState([])
+
+  const params = useQueryFilters(defaultQueryProps)
+  const filters = useProductFilters()
+
+  const {
+    mutate: deleteProductsFromSalesChannel,
+  } = useAdminDeleteProductsFromSalesChannel(salesChannelId)
+
+  const { products, count, isLoading } = useAdminProducts({
+    ...params.queryObject,
+    ...filters.queryObject,
+  })
+
+  const removeProductFromSalesChannel = (id: string) => {
+    deleteProductsFromSalesChannel({ product_ids: [{ id }] })
+  }
+
+  const removeSelectedProducts = async () => {
+    await deleteProductsFromSalesChannel({
+      product_ids: selectedRowIds.map((id) => ({ id })),
+    })
+    setSelectedRowIds([])
+  }
```

Comment:
Issue: The view is not updated when pressing remove, it requires a hard refresh before the change shows.

Reply:
yes this is currently TODO because the logic for fetching products will change